v0/iterable/transform.js

```diff
@@ -0,0 +1,190 @@
+      } else if (
+        destination.Config.trackNamedPages &&
+        message.properties.name
+      ) {
+        rawPayload = setValues(rawPayload, message, mappingJson);
+      }
+      if (destination.Config.mapToSingleEvent) {
+        rawPayload.eventName = "Loaded a Screen";
+      } else {
+        rawPayload.eventName += " screen";
+      }
+      rawPayload.createdAt = parseInt( (new Date(rawPayload.createdAt).getTime()/1000));
+      break;
+    case "track":
+      rawPayload = setValues(rawPayload, message, mappingJson);
+      rawPayload.createdAt = parseInt( (new Date(rawPayload.createdAt).getTime()/1000));
+      break;
+    case "trackPurchase":
+      rawPayload = setValues(rawPayload, message, mappingJson);
+
+      mappingJson = mappingConfig[ConfigCategory.IDENTIFY.name];
+      rawPayload.preferUserId = true;
+      rawPayload.mergeNestedObjects = true;
+      rawPayloadUser = {};
+      rawPayload.user = setValues(rawPayloadUser, message, mappingJson);
+
+      mappingJson = mappingConfig[ConfigCategory.PRODUCT.name];
+      rawPayloadItemsArray = [];
+      rawPayloadItems = {};
+
+      for (var i in message.properties.products) {
+        rawPayloadItemsArray.push(
+          setValues(
+            rawPayloadItems,
+            message.properties.products[i],
+            mappingJson
+          )
+        );
+        rawPayloadItemsArray[i].categories = rawPayloadItemsArray[
+          i
+        ].categories.split(",");
+        rawPayloadItemsArray[i].price = parseInt(rawPayloadItemsArray[i].price)
+        rawPayloadItemsArray[i].quantity = parseInt(rawPayloadItemsArray[i].quantity)
+      }
+      rawPayload.items = rawPayloadItemsArray;
+      rawPayload.createdAt = parseInt( (new Date(rawPayload.createdAt).getTime()/1000));
+      rawPayload.total = parseInt(rawPayload.total)
+     
+
+      break;
+
+    default:
+      throw Error("not supported");
+  }
+
+  return removeUndefinedValues(rawPayload);
+}
+
+function responseBuilderSimple(message, category, destination) {
+  const response = defaultRequestConfig();
+  response.endpoint = category.endpoint;
+  response.method = defaultPostRequestConfig.requestMethod;
+
+  response.body.JSON = constructPayload(message, category, destination);
+  response.userId = message.userId;
+  response.headers = {
+    "Content-Type": "application/json",
+    api_key: destination.Config.apiKey,
+  };
+  return response;
+}
+
+function processSingleMessage(message, destination) {
+  const messageType = message.type.toLowerCase();
+ 
+  var category = {};
+  switch (messageType) {
+    case EventType.IDENTIFY:
+      category = ConfigCategory.IDENTIFY;
+      break;
+    case EventType.PAGE:
+      category = ConfigCategory.PAGE;
+      break;
+    case EventType.SCREEN:
+      category = ConfigCategory.SCREEN;
+      break;
+    case EventType.TRACK:
+      const event = message.event.toLowerCase();
+      switch (event) {
+        case "order completed":
```

Comment:
should we also handle Product Added and Product Removed events ?
https://api.iterable.com/api/docs#commerce_trackPurchase

If we implement we can mention like segment, that the entire product array in the cart should be send, as Iterable doesn't have add/remove from cart support
https://segment.com/docs/connections/destinations/catalog/iterable/#ecommerce

Reply:
yes I did not handle product add/removed because in the docs I have seen that only the added or removed products are being sent.
we would need the entire cart to be able to do this.
I will add this.

Reply:
Added